src/main/java/seedu/taassist/model/TaAssist.java

```
@@ -94,6 +107,32 @@ public void removeStudent(Student key) {
         students.remove(key);
     }
 
+    //// class-level operations
+
+    /**
+     * Returns true if a class with the same identity as {@code moduleClass} exists in TA-Assist.
+     */
+    public boolean hasModuleClass(ModuleClass moduleClass) {
+        requireNonNull(moduleClass);
+        return moduleClasses.contains(moduleClass);
+    }
+
+    /**
+     * Adds a class to TA-Assist.
+     * The class must not already exist in TA-Assist.
+     */
+    public void addModuleClass(ModuleClass c) {
+        moduleClasses.add(c);
+    }
```

Maybe add `requireNonNull(c)` here to fail early rather than fail at `add(c)`

---

I have added all the `requireNonNull` for now, but we might need to look into it again since there are multiple checks across the program. There is 1 in `ModelManager` and 1 more in `TaAssist`.

---

@Bubbl3T I think it is good to have them all. Because in future we could be calling these from a new class we created.